test/unit/services/synchronization/nowait_lock_service_test.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Synchronization::NowaitLockServiceTest < ActionDispatch::IntegrationTest
+  attr_reader :lock_key
+
+  setup do
+    @lock_key = SecureRandom.uuid
+  end
+
+  test "locks then cannot lock but can lock after unlock" do
+    assert Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    assert_not Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    assert_equal 1, Synchronization::UnsafeUnlockService.call(lock_key).result
+    assert_equal 0, Synchronization::UnsafeUnlockService.call(lock_key).result
+    assert Synchronization::NowaitLockService.call(lock_key, timeout: 100).result
+  end
```

Comment:
For this test we rely on `UnsafeUnlockService` which is not tested, and a bug in `UnsafeUnlockService` can break this test. Besides, we also rely on Redis here. For those reasons this seems more an integration than a unit test for me. I think it'd be beter to mock Redis calls and split this test into several like `key exist in redis`, `key doesn't exist in redis` and `key exist but with a different value`

Reply:
correct, need to move it to integration. I don't find much value in mocking redis because I care about overall functionality. I don't care whether service performs this or that method call that is an implementation detail. Similarly the lock and unlock services need to work well together with each other. I see no value in mocking everything to test them individually that any refactoring will break. 

This may even mean that if service changes implementation to RDBMS or something completely different, for tests it will be transparent.